superset-frontend/packages/superset-ui-core/src/query/types/Datasource.ts

```diff
@@ -47,11 +47,17 @@ export interface Datasource {
   };
 }
 
-export const DEFAULT_METRICS = [
+export const DEFAULT_METRICS: Metric[] = [
   {
     metric_name: 'COUNT(*)',
     expression: 'COUNT(*)',
   },
 ];
 
+export const isValidDatasourceType = (datasource: DatasourceType) =>
+  datasource === DatasourceType.Dataset ||
```

Comment:
can we do something like `datasource in DatasourceType` or do we need to list out all of the types?